app/models/concerns/katalyst/content/container.rb

```diff
@@ -49,8 +49,24 @@ def inactive
                  dependent:  :destroy,
                  validate:   true
 
+        # Rails 7.2 dropped support for ActiveModel enums. This is a temporary workaround.
+        # Intended behaviour:
+        # attribute :state, :string
+        # enum :state, %i[published draft unpublished].index_with(&:to_s)
         attribute :state, :string
-        enum :state, %i[published draft unpublished].index_with(&:to_s)
+
+        # Returns true if there is a published version. Note that this includes drafts.
+        def published?
+          %w[published draft].include?(state)
+        end
+
+        def draft?
+          state == "draft"
+        end
+
+        def unpublished?
+          state == "unpublished"
+        end
```

Comment:
(opt) something that's been very popular of late in gus:

```
delegate :draft, :unpublished, to: :state

def state
    (super || "").inquiry
end

def published? = state.published? || state.draft?
```

Reply:
I don't really see the benefit. Saves characters? It's a bit hard to understand imo.